app/lib/common/widgets/html_editor/components/mention_list.dart

```diff
@@ -0,0 +1,289 @@
+        });
+        _scrollToSelected();
+        return KeyEventResult.handled;
+
+      case LogicalKeyboardKey.backspace:
+        final selection = widget.editorState.selection;
+        if (selection == null) return KeyEventResult.handled;
+
+        final node = widget.editorState.getNodeAtPath(selection.end.path);
+        if (node == null) return KeyEventResult.handled;
+
+        // Get text before cursor
+        final text = node.delta?.toPlainText() ?? '';
+        final cursorPosition = selection.end.offset;
+
+        if (_canDeleteLastCharacter()) {
+          // Check if we're about to delete an mention symbol
+          if (cursorPosition > 0 &&
+              text[cursorPosition - 1] ==
+                  MentionType.toStr(widget.mentionType)) {
+            widget.onDismiss(); // Dismiss menu when is deleted
+          }
+          widget.editorState.deleteBackward();
+        } else {
+          // Workaround for editor regaining focus
+          widget.editorState.apply(
+            widget.editorState.transaction..afterSelection = selection,
+          );
+        }
+        return KeyEventResult.handled;
+
+      default:
+        if (event.character != null &&
+                !HardwareKeyboard.instance.isAltPressed &&
+                !HardwareKeyboard.instance.isMetaPressed ||
+            !HardwareKeyboard.instance.isShiftPressed) {
+          widget.editorState.insertTextAtCurrentSelection(event.character!);
+          return KeyEventResult.handled;
+        }
+        return KeyEventResult.ignored;
+    }
+  }
+
+  void _selectItem(String id, String displayName) {
+    final selection = widget.editorState.selection;
+    if (selection == null) return;
+
+    final transaction = widget.editorState.transaction;
+    final node = widget.editorState.getNodeAtPath(selection.end.path);
+    if (node == null) return;
+
+    final text = node.delta?.toPlainText() ?? '';
+    final cursorPosition = selection.end.offset;
+
+    // Find the trigger symbol position by searching backwards from cursor
+    int atSymbolPosition = -1;
+    for (int i = cursorPosition - 1; i >= 0; i--) {
+      if (text[i] == MentionType.toStr(widget.mentionType)) {
+        atSymbolPosition = i;
+        break;
+      }
+    }
+
+    if (atSymbolPosition == -1) return; // No trigger found
+
+    // Calculate length from trigger to cursor
+    final lengthToReplace = cursorPosition - atSymbolPosition;
+
+    transaction.replaceText(
+      node,
+      atSymbolPosition, // Start exactly from trigger
+      lengthToReplace, // Replace everything including trigger
+      ' ',
+      attributes: {
+        MentionBlockKeys.mention: {
+          MentionBlockKeys.type: widget.mentionType.name,
+          if (widget.mentionType == MentionType.user)
+            MentionBlockKeys.userId: id
+          else
+            MentionBlockKeys.roomId: id,
+          MentionBlockKeys.displayName: displayName,
+        },
+      },
```

Comment:
so ... `attributes`  Attributes = [Map](https://api.flutter.dev/flutter/dart-core/Map-class.html)<[String](https://api.flutter.dev/flutter/dart-core/String-class.html), `dynamic`>, right? so, why not have the "mention"-string-key point to a _typed_ object. all those `MentionBlockKeys`-with strings make it really hard to follow and easy to mess up. I'd rather have a typed:

```suggestion
      attributes: {
        MENTIONS_KEY: widget.mentionType == MentionType.user ? 
            MentionAttributes.user(userId: id, displayName: displayName)
            MentionAttributes.room(roomId: id, displayName: displayName)
      },
```

For a type `MentionAttributes`, along the lines of (pseudocode):

```
class MentionAttributes {
  final String? userId;
  final String? roomId;
  final String? displayName;
  final MentionType type;
  
  const MentionAttributes({this.userId, this.roomId, required this.type, this.displayName});

  factory MentionAtrributes.user(required this.userId, {this.displayName}) {
     type = MentionType.user;
  }
}
```